### ISSUE TITLE
command block permissions for /zone command

### ISSUE THREAD

**Issue by zwrm1** [ForgeEssentials/ForgeEssentials#2077]
hey guys, i was attempting to do something like "/zone exit zoneid message" via a command block its returning "you do not have permission to use this command" any thoughts?

($COMMANDBLOCK seems to have * granted in the permissions file, and using build 1167)

**Comment by olee** [comment]
Try enabling `/p debug` and try again :wink:

**Comment by spacebuilder2020** [comment]
I have noticed this also happens when trying to run commands that should be run by a player.  As a recall, command blocks are not a fake player but are actually extensions of console.

**Comment by zwrm1** [comment]
I have tried /p debug but it doesn't give any output on which permissions might be being checked (almost like it's not checking any). 

BTW I am a developer (20 years commercial experience) and I have an interest in contributing to this project on a small scale (not as a lead but perhaps learning enough to fix a bug or two, if you are interested :)

**Comment by olee** [comment]
Regarding development: sounds good. You should say hi on our IRC.

As for the issue: You need to use `/p debug` as a player and you need to stand somewhere around the command block or it will not show the permission check. If that's what you did, I might have to check it myself....

**Comment by zwrm1** [comment]
Yes that's what did, I was standing next to it clicking a button attached to the command block. I have attached screen shots so you can see the setup and the output. Thanks for your assistance with this, btw, as well as your work in general. I can tell be reading the posts here that you contribute a significant amount of your time :)

![Uploading 20160511_180259.jpg…]()

**Comment by zwrm1** [comment]
![20160511_180259](https://cloud.githubusercontent.com/assets/19308142/15199471/fb79ad9e-17a3-11e6-82ec-eff7c6411fb3.jpg)
![20160511_180210](https://cloud.githubusercontent.com/assets/19308142/15199474/0010abf0-17a4-11e6-9156-ad1b35dec0b8.jpg)

**Comment by zwrm1** [comment]
this seems to affect some other command block commands as well (maybe many...) for example /morph demorph also produces the same output. I think this might be a very general problem with the way command block permissions are handled.

**Comment by zwrm1** [comment]
also, this may be affecting scripting as well, as I have tried to make some scripted commands to do the same things, but get similar permission errors.

**Comment by spacebuilder2020** [comment]
This is a issue in multiple places in the code.  A lot of methods either fail if the sender is not an entity player or only consider a class of instance MinecraftServer as a console based sender.  The change I made in PR #2088 includes some code that should fix the issue with command blocks, etc in scripting as long as you give the command block sender access to the command you want it to run in the permissions.  Since this is happening in multiple places, there are a fail amount of fixes that need to be done.
